src/org/zaproxy/zap/extension/domxss/TestDomXSS.java

```diff
@@ -510,35 +515,33 @@ public void scan() {
 					}
 				}
 			}
+			super.scan();
 		} finally {
-			this.returnFirefoxDriver(fxDriver);
+			this.returnFirefoxDriver(ffDriver);
 		}
 	}
 
-	public boolean scan(ArrayList<WebDriverWrapper> drivers, String attackVector) {
-    	HttpMessage msg = getBaseMsg();
-    	String url = msg.getRequestHeader().getURI().toString();
-    	String currURL = url + attackVector;
+	public boolean scan(WebDriverWrapper driver, String attackVector, String currUrl) {
+    	HttpMessage msg = getNewMsg();
     	
-    	for (WebDriverWrapper driver : drivers) {
-        	try {
-    			scanHelper(driver, attackVector, currURL);
-    		} catch (DomAlertException e) {
-    			String tagName = e.getTagName();
-    			String otherInfo = "";
-    			if (tagName != null) {
-    				otherInfo = "Tag name: " + tagName + 
-    						" Att name: " + e.getAttributeName() +
-    						" Att id: " + e.getAttributeId();
-    			}
-    			
-    			bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, 
-    					e.getUrl(), null, e.getAttack(), 
-    					otherInfo, null, msg);
-    			Stats.incCounter("domxss.attack." + attackVector);
-    			return true;
-			}
-    	}
+
+        try {
+    		scanHelper(driver, attackVector, currUrl);
```

Comment:
Could use the driver directly in `scanHelper` (and remove the driver parameters).